.github/actions/package-macos/action.yaml

```diff
@@ -10,141 +10,98 @@ inputs:
-          *arm64)
-            echo "SELF_HOSTED=true" >> $GITHUB_OUTPUT
-            ;;
-          *)
-            echo "SELF_HOSTED=false" >> $GITHUB_OUTPUT
-            ;;
-        esac
+        curl -LO https://cmake.org/files/v3.31/cmake-3.31.7-macos-universal.tar.gz
+        tar -xzf cmake-3.31.7-macos-universal.tar.gz
+        sudo mv cmake-3.31.7-macos-universal/CMake.app/Contents/bin/cmake /usr/local/bin
+        sudo rm -rf cmake-3.31.7-macos-universal
+        cmake --version
     - uses: actions/cache@d4323d4df104b026a6aa633fdb11d772146be0bf # v4.2.2
       id: cache
-      if: steps.prepare.outputs.SELF_HOSTED != 'true'
       with:
-        path: ${{ steps.prepare.outputs.OTP_INSTALL_PATH }}
-        key: otp-install-${{ inputs.otp }}-${{ inputs.os }}-static-ssl-disable-hipe-disable-jit-20240524-1
+        path: ~/.kerl/installations/${{ inputs.otp }}
+        key: otp-install-${{ inputs.otp }}-${{ inputs.os }}-disable-jit-20250325
     - name: build erlang
       if: steps.cache.outputs.cache-hit != 'true'
       shell: bash
+      env:
+        OTP_VERSION: ${{ inputs.otp }}
       run: |
-        OTP_SOURCE_PATH="${{ steps.prepare.outputs.OTP_SOURCE_PATH }}"
-        OTP_INSTALL_PATH="${{ steps.prepare.outputs.OTP_INSTALL_PATH }}"
-        SELF_HOSTED="${{ steps.prepare.outputs.SELF_HOSTED }}"
-        # when it's self-hosted, it never hits the cache,
-        # skip rebuild if it's self-hosted and the install path already has a 'bin'
-        if [ "${SELF_HOSTED:-false}" = 'true' ]; then
-          if [ -n "$OTP_INSTALL_PATH" ] && [ -d "$OTP_INSTALL_PATH/bin" ]; then
-            echo "Skip rebuilding OTP, found $OTP_INSTALL_PATH"
-            exit 0
-          fi
-        fi
-        ## when it's not self-hosted, or the install path is not found,
-        ## build otp from source code.
-        if [ -d "$OTP_SOURCE_PATH" ]; then
-          rm -rf "$OTP_SOURCE_PATH"
-        fi
-        git clone --depth 1 --branch OTP-${{ inputs.otp }} https://github.com/emqx/otp.git "$OTP_SOURCE_PATH"
-        cd "$OTP_SOURCE_PATH"
-        if [ "$(arch)" = arm64 ]; then
-            ODBCHOME="$(brew --prefix unixodbc)"
-            export CFLAGS="-O2 -g -I${ODBCHOME}/include"
-            export LDFLAGS="-L${ODBCHOME}/lib"
-            WITH_ODBC="--with-odbc=${ODBCHOME}"
-        else
-            WITH_ODBC=""
-        fi
-        ./configure --disable-dynamic-ssl-lib --with-ssl=$(brew --prefix openssl@1.1) ${WITH_ODBC} --disable-hipe --disable-jit --prefix="$OTP_INSTALL_PATH"
-        make -j$(nproc)
-        rm -rf "$OTP_INSTALL_PATH"
-        make install
-        if [ "$(arch)" = arm64 ]; then
-            unset CFLAGS
-            unset LDFLAGS
-        fi
+        export LDFLAGS="-L$(brew --prefix unixodbc)/lib"
+        export CC="/usr/bin/gcc -I$(brew --prefix unixodbc)/include"
+        export KERL_CONFIGURE_OPTIONS="--disable-jit --with-ssl=$(brew --prefix openssl) --with-odbc=$(brew --prefix unixodbc)"
```

Comment:
FYI, last time we have dynamic linked ssl, it brokes. as the SSL deps are in brew local dir which are so different from machine to machine.

Reply:
let me get it back